Feature/49951 meetings tab on work package page

jjabari-op opened the pull request:
https://community.openproject.org/work_packages/49951

**TODOS** 

- [x] Dialog prevents CKEditor toolbar from opening correctly (it assumes the invalid width and collapses). @jonasjabari  I'd assume this gets fixed if you lazy load the dialog content, which we should do anyway.
<img width="977" alt="Bildschirmfoto 2023-09-21 um 20 20 32" src="https://github.com/opf/openproject/assets/459462/f9999024-5c9c-4c87-9ef1-c0c05c2b05d0">

jjabari-op committed:
[#49951] Meetings tab on work package page

https://community.openproject.org/work_packages/49951

jjabari-op committed:
added meetings tab for work package page as and add-on feature for #13725, specs missing

jjabari-op committed:
Merge branch 'dev' into feature/49951-meetings-tab-on-work-package-page

jjabari-op committed:
cleanup and updated to-do comments

jjabari-op committed:
cleanup left-overs of include child wps draft

oliverguenther committed:
Use autocomplete select decoration for meeting selector

oliverguenther committed:
Use proper prefix for turbo frame src and add content-loader example

jjabari-op committed:
WIP: Adding specs for meetings tab, specs for adding agenda items through meetings tab missing

jjabari-op committed:
Merge branch 'feature/49951-meetings-tab-on-work-package-page' of github.com:opf/openproject into feature/49951-meetings-tab-on-work-package-page

jjabari-op committed:
re-enabled temporarly disabled specs

jjabari-op committed:
cleanup redundant UI update

m-graf commented on modules/meeting/app/components/work_package_meetings_tab/add_work_package_to_meeting_form_component.rb:
```diff
@@ -0,0 +1,97 @@
+          url: work_package_meeting_agenda_items_path(@work_package)
+        ) do |f|
+          component_collection do |collection|
+            collection.with_component(Primer::Alpha::Dialog::Body.new(test_selector: 'op-add-work-package-to-meeting-dialog-body')) do
```
Might be cool to extract these selectors to constants

m-graf reviewed (commented):
Cool :) Just some comments

jjabari-op committed:
implemented potentiallly reuseable async primer dialog, using turbo frame and a bit of stimulus

jjabari-op committed:
cleanup

jjabari-op committed:
finalized meeting tabs specs

jjabari-op committed:
adjusted contracts towards usage in meetings tab

jjabari-op committed:
Merge branch 'dev' into feature/49951-meetings-tab-on-work-package-page

oliverguenther committed:
Do not output meetings tab on new work packages

oliverguenther committed:
Don't use through association to check for presence, use meeting itself

oliverguenther committed:
Add TurboElement type to prevent any

oliverguenther committed:
Rework into view

oliverguenther committed:
Visit response in case frame is missing

If you're setting up the wrong frame response, you get "Content missing" by default. This is not very helpful for identifying the error.

We can listen to the turbo frame-missing event and promote it to a full visit instead

https://community.openproject.org/work_packages/50210

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/49951-meetings-tab-on-work-package-page

oliverguenther committed:
Add nowrap to "Added by"

oliverguenther committed:
Unify "added by" display

oliverguenther committed:
Adapt create contract spec for visibility check

oliverguenther committed:
Use contract to get assignable meetings

oliverguenther committed:
Use explicit option to set decorated status, even if options empty
